src/backend/distributed/utils/metadata_cache.c

```diff
@@ -677,6 +722,32 @@ InvalidateDistRelationCacheCallback(Datum argument, Oid relationId)
 			cacheEntry->isValid = false;
 		}
 	}
+
+	/* now check whether we're in the middle of dropping the extension */
+
+	/* if we're not in a transaction then we'll crash while looking up oids */
+	inTransaction = IsTransactionState();
+	if (!inTransaction)
+	{
+		return;
```

Comment:
For those calls that means there's no invalidation. This is very handwavey but in every situation I tried pg_dist_shard_partition is invalidated at least once from within a transaction during the DROP.